xbmc/windowing/gbm/GBMUtils.cpp

```diff
@@ -45,16 +75,36 @@ bool CGBMUtils::InitGbm(struct gbm *gbm, int hdisplay, int vdisplay)
   return true;
 }
 
-void CGBMUtils::DestroyGbm(struct gbm *gbm)
+void CGBMUtils::DestroySurface(struct gbm *gbm)
 {
-  if(gbm->surface)
+  if (!gbm->surface)
+    CLog::Log(LOGWARNING, "CGBMUtils::%s - surface already destroyed", __FUNCTION__);
+
+  if (gbm->surface)
```

Comment:
You should also check if gbm itself is not a nullptr.

Reply:
There's a lot more spots that check gbm->someptr, the question is if you prefer to hard fail in all those cases or add if (!gbm) checks everywhere.

Reply:
Passing `gbm` around should not be needed in the first place.
Original intent was to merge the `struct gbm` with `CGBMUtils` and convert all static methods to member methods, but did not get that far.